packages/create-cloudflare/src/common.ts

```diff
@@ -591,15 +578,25 @@ function prepareCommitMessage(commitMessage: string): string {
 }
 
 export function quoteShellArgs(args: string[]): string {
-	if (process.platform !== "win32") {
-		return quote(args);
-	} else {
+	if (process.platform === "win32") {
 		// Simple Windows command prompt quoting if there are special characters.
 		const specialCharsMatcher = /[&<>[\]|{}^=;!'+,`~\s]/;
 		return args
 			.map((arg) =>
 				arg.match(specialCharsMatcher) ? `"${arg.replaceAll(`"`, `""`)}"` : arg
 			)
 			.join(" ");
+	} else {
+		return args
+			.map((s) => {
+				if (/["\s]/.test(s) && !/'/.test(s)) {
+					return "'" + s.replace(/(['\\])/g, "\\$1") + "'";
+				}
+				if (/["'\s]/.test(s)) {
+					return '"' + s.replace(/(["\\$`!])/g, "\\$1") + '"';
+				}
+				return s;
+			})
+			.join(" ");
```

Comment:
I think this match & replace logic is really not robust and unlikely to be a correct generic solution

For example see this example test I made: 
[![Screenshot 2024-01-25 at 14 18 23](https://github.com/cloudflare/workers-sdk/assets/61631103/ea718c60-0173-4180-b43b-58c9ccc204ef)](https://stackblitz.com/edit/typescript-6r74kz?file=index.ts)


Unless I'm missing something?

Reply:
I think this is actually correct.
- If it contains a single quote then it must be wrapped in double quotes. 
- if it contains a double quote then it must be wrapped in single quotes
- If it contains whitespace then it must be wrapped in single or double quotes.

Reply:
Cool thanks 😄